src/authenticator.ts

```diff
@@ -166,12 +166,17 @@ export class Authenticator<User = unknown> {
     request: Request,
     options: { successRedirect?: never; failureRedirect: string }
   ): Promise<User>;
+  async isAuthenticated(
+    request: Request, 
+    options: { successRedirect: string; failureRedirect: string }
+  ): Promise<User>;
```

Comment:
This should be void not User, in case the user was authenticated it would redirect so you never really get anything back